samples/crypto/tls/src/tls_functions_client.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(tls_functions);
+
+#include <nrf.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <net/socket.h>
+#include <net/net_core.h>
+#include <net/tls_credentials.h>
+#include <linker/sections.h>
+
+#include "certificate.h"
+#include "psa_tls_functions.h"
+#include "psa_tls_credentials.h"
+
+char m_recv_buffer[RECV_BUFFER_SIZE]; /* Buffer for storing payload */
+
+/* Define thread parameters for TLS Client. */
+static int setup_tls_client_socket(void);
+static void process_tls_client(void);
+K_THREAD_DEFINE(tls_client_thread_id, THREAD_STACK_SIZE, process_tls_client,
+		NULL, NULL, NULL, THREAD_PRIORITY, 0, -1);
+
+
+void tls_start_thread(void)
+{
+	k_thread_start(tls_client_thread_id);
+}
+
+/** @brief Function for sending all data in a given buffer to a connected
+ * socket.
+ *
+ * @param   sock        The socket identifier to send data to.
+ * @param   buf         Pointer to the data buffer.
+ * @param   len         Number of bytes to send from buf.
+ *
+ * @returns 0 on success.
+ */
+static ssize_t send_buffer(int sock, const uint8_t *buf, size_t len)
+{
+	ssize_t sent_len;
+
+	while (len > 0) {
+		sent_len = send(sock, buf, len, 0);
+		if (sent_len < 0) {
+			return sent_len;
```

Comment:
I think it'd be nice to return `-errno` here, so it can be logged later.

Reply:
fixed